tst/com/amazon/corretto/crypto/provider/test/HpkeCipherTest.java

```diff
@@ -0,0 +1,104 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider.test;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import com.amazon.corretto.crypto.provider.HpkeParameterSpec;
+import java.security.*;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.spec.ECGenParameterSpec;
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.junit.jupiter.api.parallel.ResourceAccessMode;
+import org.junit.jupiter.api.parallel.ResourceLock;
+
+@ExtendWith(TestResultLogger.class)
+@Execution(ExecutionMode.CONCURRENT)
+@ResourceLock(value = TestUtil.RESOURCE_GLOBAL, mode = ResourceAccessMode.READ)
+public class HpkeCipherTest {
+  private KeyPairGenerator getGenerator() throws GeneralSecurityException {
+    return KeyPairGenerator.getInstance("HPKE", TestUtil.NATIVE_PROVIDER);
+  }
+
+  private static Cipher getCipher() throws GeneralSecurityException {
+    return Cipher.getInstance("HPKE", TestUtil.NATIVE_PROVIDER);
+  }
+
+  @Test
+  public void testWrapUnwrap() throws GeneralSecurityException {
+    final HpkeParameterSpec[] namedSpecs = {
```

Comment:
Check out JUnit's built-in support for parameterized testing.

https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests